shells/pipes-shell/source/serialize-verb.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {logsFactory} from '../../../build/platform/logs-factory.js';
+
+const {log} = logsFactory('SerializeVerb');
+
+// TODO(sjmiles): allow serializing verbs to reduce synchronization
+// burden on apps. Not strictly part of runtime api: revisit at some point.
+
+// have separate queues per verb
+const work = {};
+
+// only for logging
+let index = 0;
+
+export const serializeVerb = async (verb, asyncTask) => {
+  // only for logging
+  let id = ++index;
+  // capture previous work-promise
+  const busy = work[verb];
+  // construct new work-promise
+  let resolve;
+  work[verb] = new Promise(_resolve => resolve = _resolve);
+  // wait for previous work (yields at await)
+  log(`awaiting [${verb}][${id-1}]`);
+  await busy;
+  // begin new work (yields at await)
+  log(`invoking [${verb}][${id}]`);
+  await (async () => {
+    const result = await asyncTask();
```

Comment:
shall we add `await arc.idle;` in run-arc implementation maybe? otherwise i think the promise here might be resolved before the recipe was done instantiating. WDYT?

Reply:
I do have general concerns about a lack of strong definitions. Arcs don't have _completion states_ at all because they are reactive. It's not clear to me how to characterize _the point at which an Arc first becomes idle_.
Having said that, this whole PR is a stop-gap (IMO), so I'm not averse to adding the await you suggest.